packages/gatsby/src/schema/graphql-engine/bundle-webpack.ts

```diff
@@ -93,7 +104,7 @@ export async function createGraphqlEngineBundle(
             },
             {
               // specific set of loaders for gatsby-node files - our babel transform that removes lifecycles not needed for engine -> relocator
-              test: /gatsby-node\.([cm]?js)$/,
+              test: /gatsby-node\.(cjs|mjs|js|ts)$/,
```

Comment:
previously our test was skipping the removal of unrelated to engines exports/lifecycles for `gatsby-node.ts`